builder/azure/arm/step_capture_image.go

```diff
@@ -70,6 +68,18 @@ func (s *StepCaptureImage) captureImage(ctx context.Context, vmId hashiVMSDK.Vir
 	return nil
 }
 
+func (s *StepCaptureImage) getVMID(ctx context.Context, vmId hashiVMSDK.VirtualMachineId) (string, error) {
+	vmResponse, err := s.client.VirtualMachinesClient.Get(ctx, vmId, hashiVMSDK.DefaultGetOperationOptions())
+	if err != nil {
+		return "", err
+	}
+	if vmResponse.Model != nil {
```

Comment:
What does it mean for vmResponse.Model to be nil? 

It doesn't seem like its an error and having an empty string is a valid response for this step. Is that correct? 

To make this a little more explicit I would suggest a small refactor
```
if vmResponse.Model == nil {
  // an empty VMID is value when ....
  return "", nil
}
vmId := vmResponse.Model.Properties.VMId
return *vmId, nil
```

Reply:
Good call, I think I just wrote this wrong tbh if the Model is nil we just need to error here with the same error we do everywhere

Reply:
Implemented this in new commit, not counting the general SDK error, but the Model being nill here IMO should just throw an error, if we can't Get the VM here we won't be able to capture it later on anyways